katsdpcontroller/product_config.py

```diff
@@ -315,12 +315,14 @@ def __init__(self, name: str, src_streams: Sequence[Stream], *,
                  adc_sample_rate: float,
                  centre_frequency: float,
                  band: str,
-                 antenna: katpoint.Antenna) -> None:
+                 antenna: katpoint.Antenna,
+                 command_line_extra: Iterable[str] = ()) -> None:
         super().__init__(name, [])
         self.adc_sample_rate = adc_sample_rate
         self.centre_frequency = centre_frequency
         self.band = band
         self.antenna = antenna
+        self.command_line_extra = list(command_line_extra)
```

Comment:
An alternative would be to make the default value be an empty list, but linting tools tend to complain about mutable default values (even though it's safe in this case since it's not mutated). Or I could make `self.command_line_extra` be a tuple, but then I'd have to cast the list that comes out of the JSON parser to a tuple. Or I could make the default be None, but then I have to do something like `command_line_extra if command_line_extra else []` which I find less readable.

Reply:
All fair points. Probably easiest to leave it as you've done it.